### PR TITLE
exoscale: destroy the SSH keypair when removing the machine

### PR DESCRIPTION
**Pull request by vincentbernat**
Otherwise, when recreating a machine with the same name, we would get an
error about this SSH keypair already existing.

To avoid any confusion, let's say that machine will handle all keypairs
and the user cannot provide its own. Therefore, move the keypairs into
their own "namespace" and remove the option to specify a keypair.

This is also what is done by Digital Ocean.

First commit is totally unrelated but I didn't want to do a PR just for it. Second commit is needed to get the appropriate method to destroy keypair.

### CODE REVIEW

File: Godeps/_workspace/src/github.com/pyr/egoscale/src/egoscale/types.go [hunk 1]
```diff
@@ -12,13 +12,17 @@ type Client struct {
 	apiSecret string
 }
 
-
 type Error struct {
 	ErrorCode int `json:"errorcode"`
```
Comment (ehazlett) [inline Godeps/_workspace/src/github.com/pyr/egoscale/src/egoscale/types.go]: This looks like it needs a `go fmt`.

File: drivers/exoscale/exoscale.go [hunk 2]
```diff
@@ -383,6 +376,14 @@ func (d *Driver) Stop() error {
 
 func (d *Driver) Remove() error {
 	client := egoscale.NewClient(d.URL, d.ApiKey, d.ApiSecretKey)
+
+	// Destroy the SSH key
+	_, err := client.DeleteKeypair(d.KeyPair)
```
Comment (ehazlett) [inline drivers/exoscale/exoscale.go]: Minor: fyi you can do this as:
```
if _, err := client.DeleteKeypair(d.KeyPair); err != nil {
        return err
}
```

File: Godeps/_workspace/src/github.com/pyr/egoscale/src/egoscale/types.go (hunk 1 above)
Comment (vincentbernat) [inline Godeps/_workspace/src/github.com/pyr/egoscale/src/egoscale/types.go]: That's "upstream". `go fmt` would also change more than that. I'll ask.

**Comment by vincentbernat** [comment]
Updated for both comments. Integration tests are successful.

File: Godeps/_workspace/src/github.com/pyr/egoscale/src/egoscale/types.go (hunk 1 above)
Comment (ehazlett) [inline Godeps/_workspace/src/github.com/pyr/egoscale/src/egoscale/types.go]: Ah ok -- I didn't check the path.  Thanks.

**Comment by ehazlett** [comment]
LGTM

**Comment by vincentbernat** [comment]
Thanks!

**Comment by ehazlett** [comment]
yw!

**Pull request merged**